src/server_name.rs

```diff
@@ -49,6 +51,18 @@ impl<'a> ServerName<'a> {
             Self::IpAddress(i) => ServerName::IpAddress(*i),
         }
     }
+
+    /// Return the string representation of this `ServerName`.
```

Comment:
I think it would be nice to be explicit about the conditions under which this allocates (which are a little unintuitive).

Reply:
I added some explicit documentation about this. Thanks!